helm/hpcc/templates/_helpers.tpl

```diff
@@ -193,36 +192,35 @@ Pass in root and includeLabels (optional)
   {{- end }}
  {{- end }}
 {{- end }}
+{{- /*Backwards compatibility: when no data planes defined in values, create default pvc*/ -}}
+{{- if and (or (has "data" $includeLabels) (has "" $includeLabels)) (not $dataStorage.plane) }}
+- name: datastorage
+  mountPath: "/var/lib/HPCCSystems/hpcc-data"
+{{- end }}
 {{- end -}}
 
 {{/*
 Add data volume mount
-If any storage planes are defined that name pvcs they will be mounted
+Pass in root
 */}}
 {{- define "hpcc.addDataVolumeMount" -}}
-{{- $storage := (.root.Values.storage | default dict) -}}
-{{- $dataStorage := ($storage.dataStorage | default dict) -}}
-{{- $includeLabels :=  append (.includeLabels | default list) "data" | uniq -}}
-{{- include "hpcc.addVolumeMounts"  (dict "root" .root "includeLabels" $includeLabels) -}}
-{{- if (not $dataStorage.plane) }}
-- name: datastorage
-  mountPath: "/var/lib/HPCCSystems/hpcc-data"
-{{- end }}
+{{- include "hpcc.addVolumeMounts"  (dict "root" .root "includeLabels" (list "data" "")) -}}
 {{- end -}}
 
 {{/*
-Add volume
+Add volumes
 Pass in root and includeLabels (optional)
 */}}
 {{- define "hpcc.addVolumes" -}}
 {{- /*Create local variables which always exist to avoid having to check if intermediate key values exist*/ -}}
 {{- $storage := (.root.Values.storage | default dict) -}}
+{{- $dataStorage := ($storage.dataStorage | default dict) -}}
 {{- $planes := ($storage.planes | default list) -}}
 {{- $includeLabels := .includeLabels -}}
 {{- range $plane := $planes -}}
  {{- if $plane.pvc -}}
   {{- $matchedLabels := include "hpcc.doesStorageLabelsMatch" (dict "plane" $plane "includeLabels" $includeLabels) -}}
-  {{- if or (not $plane.labels) (eq $matchedLabels "true") }}
+  {{- if eq $matchedLabels "true" }}
```

Comment:
{{- if $matchedLabels }}

Reply:
or because doesStorageLabelsMatch doesn't return just "true", need to be same as line 181:
```
{{- if ne $matchedLabels "" }}
```